qa/workunits/rbd/rbd_mirror.sh

```diff
@@ -312,19 +371,34 @@ wait_for_snap_removed_from_trash ${CLUSTER1} ${PARENT_POOL} ${parent_image} ${pa
 
 testlog " - clone v2 non-primary"
 create_snapshot ${CLUSTER2} ${PARENT_POOL} ${parent_image} ${parent_snap}
+if [ "${RBD_MIRROR_MODE}" = "snapshot" ]; then
+  mirror_image_snapshot ${CLUSTER2} ${PARENT_POOL} ${parent_image}
+fi
 wait_for_snap_present ${CLUSTER1} ${PARENT_POOL} ${parent_image} ${parent_snap}
-clone_image ${CLUSTER1} ${PARENT_POOL} ${parent_image} ${parent_snap} ${POOL} \
-            ${clone_image}_v2 --rbd-default-clone-format 2
+if [ "${RBD_MIRROR_MODE}" = "snapshot" ]; then
+  clone_image_and_enable_mirror ${CLUSTER1} ${PARENT_POOL} ${parent_image} \
+	${parent_snap} ${POOL} ${clone_image}_v2 snapshot --rbd-default-clone-format 2
+else
+  clone_image ${CLUSTER1} ${PARENT_POOL} ${parent_image} ${parent_snap} ${POOL} \
+	${clone_image}_v2 --rbd-default-clone-format 2
+fi
 remove_snapshot ${CLUSTER2} ${PARENT_POOL} ${parent_image} ${parent_snap}
 test_snap_removed_from_trash ${CLUSTER2} ${PARENT_POOL} ${parent_image} ${parent_snap}
+if [ "${RBD_MIRROR_MODE}" = "snapshot" ]; then
+  mirror_image_snapshot ${CLUSTER2} ${PARENT_POOL} ${parent_image}
+fi
 wait_for_snap_moved_to_trash ${CLUSTER1} ${PARENT_POOL} ${parent_image} ${parent_snap}
 remove_image_retry ${CLUSTER1} ${POOL} ${clone_image}_v2
 wait_for_snap_removed_from_trash ${CLUSTER1} ${PARENT_POOL} ${parent_image} ${parent_snap}
 remove_image_retry ${CLUSTER2} ${PARENT_POOL} ${parent_image}
 
 testlog "TEST: data pool"
 dp_image=test_data_pool
-create_image ${CLUSTER2} ${POOL} ${dp_image} 128 --data-pool ${PARENT_POOL}
+if [ "${RBD_MIRROR_MODE}" = "snapshot" ]; then
```

Comment:
Do you need a 'if' and 'else' here? Wouldn't the following handle both the cases?
``` 
create_image_and_enable_mirror ${CLUSTER2} ${POOL} ${dp_image} 128 --data-pool ${PARENT_POOL}
```

Reply:
Rather you might have meant `create_image_and_enable_mirror ${CLUSTER2} ${POOL} ${dp_image} ${RBD_MIRROR_MODE} 128 --data-pool ${PARENT_POOL}`